nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/registry/flow/ImportAllFlows.java

```diff
@@ -0,0 +1,331 @@
+import org.apache.nifi.toolkit.cli.impl.result.StringResult;
+import org.apache.nifi.toolkit.cli.impl.util.JacksonUtils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ImportAllFlows extends AbstractNiFiRegistryCommand<StringResult> {
+    private static final String FILE_NAME_PREFIX = "toolkit_registry_export_all_";
+    private static final String SKIPPING_BUCKET_CREATION = " already exists, skipping bucket creation...";
+    private static final String SKIPPING_IMPORT = " already exists, skipping import...";
+    private static final String SKIPPING_FLOW_CREATION = " already exists, skipping flow creation...";
+    private static final String IMPORT_COMPLETED = "Import completed...";
+    private static final String ALL_BUCKETS_COLLECTED = "All buckets collected...";
+    private static final String ALL_FLOWS_COLLECTED = "All flows collected...";
+    private static final String ALL_FLOW_VERSIONS_COLLECTED = "All flow versions collected...";
+    private static final String FILE_NAME_SEPARATOR = "_";
+    private static final String STORAGE_LOCATION_URL = "%s/nifi-registry-api/buckets/%s/flows/%s/versions/%s";
+    private static final ObjectMapper MAPPER = JacksonUtils.getObjectMapper();
+    private final ListBuckets listBuckets;
+    private final ListFlows listFlows;
+    private final ListFlowVersions listFlowVersions;
+
+    public ImportAllFlows() {
+        super("import-all-flows", StringResult.class);
+        this.listBuckets = new ListBuckets();
+        this.listFlows = new ListFlows();
+        this.listFlowVersions = new ListFlowVersions();
+    }
+
+    @Override
+    protected void doInitialize(Context context) {
+        addOption(CommandOption.INPUT_SOURCE.createOption());
+        addOption(CommandOption.SKIP_EXISTING.createOption());
+
+        listBuckets.initialize(context);
+        listFlows.initialize(context);
+        listFlowVersions.initialize(context);
+    }
+
+    @Override
+    public String getDescription() {
+        return "From a provided directory as input, the directory content must be generated by the export-all-flows command, " +
+                "based on the file contents, the corresponding buckets, flows and flow versions will be created." +
+                "If not configured otherwise, already existing objects will be skipped.";
+    }
+
+    @Override
+    public StringResult doExecute(final NiFiRegistryClient client, final Properties properties) throws IOException, NiFiRegistryException, ParseException, CommandException {
+        final boolean skip = Boolean.parseBoolean(getRequiredArg(properties, CommandOption.SKIP_EXISTING));
```

Comment:
It seems like `SKIP_EXISTING` really only relates to skipping the import of versions for a flow when the flow already exists. So say my registry already has `Flow 1` with `v1` and `v2` and now I am trying to do an import that has the same flow, but has one additional version of `v3`. If I specify `--skipExisting true` then `v3` does not get imported, and if specify `--skipExisting false` then it does get imported.

Is there a situation in one of the uses cases where you would want to use `--skipExisting true` and not get `v3` imported?

If we do keep `--skipExisting`, I am thinking it might be clearer if the way it was used was to make it a non-required argument, and if it is not specified then the behavior is as if it was `--skipExisting false`. If they do specify it, then it can be an argument with no value and specifying `--skipExisting` means `--skipExisting true`.

Default usage would just be:
```
registry import-all-flows -i /path/to/export
```

Thoughts?